include/HubFramework/HUBContentOperationContext.h

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright (c) 2016 Spotify AB.
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+#import "HUBConnectivityState.h"
+
+@protocol HUBFeatureInfo;
+@protocol HUBViewModelBuilder;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ *  Protocol defining the public API of a content operation context
+ *
+ *  This API is currently only used with `HUBBlockContentOperation`. For more information
```

Comment:
Do we intend to send these context to “normal” content operations as well? In the future that is. In other words could we unify all operations by changing:
```objc
- (void)performForViewURI:(NSURL *)viewURI
              featureInfo:(id<HUBFeatureInfo>)featureInfo
        connectivityState:(HUBConnectivityState)connectivityState
         viewModelBuilder:(id<HUBViewModelBuilder>)viewModelBuilder
            previousError:(nullable NSError *)previousError;
```

To:
```
- (void)performForContext:(id<HUBContentOperationContext>)context;
```

---

It would have the benefit of not just unifying the API. Making sure this isn’t some special on-off edge case but a first class member. As well as making it _much_ easier to introduce, deprecate and remove “parameters” in the future.